Error compiling `h5_test/tst_h_files4` on MacOS

The following error is now being observed with the latest XCode update on MacOS:

```
/Users/wfisher/Desktop/gitprojects/netcdf-c/h5_test/tst_h_files4.c:167:64: error: incompatible function pointer types passing 'herr_t (hid_t, const char *, const H5L_info2_t *, void *)' (aka 'int (long long, const char *, const H5L_info2_t *, void *)') to parameter of type 'H5L_iterate1_t' (aka 'int (*)(long long, const char *, const H5L_info1_t *, void *)') [-Wincompatible-function-pointer-types]
         if (H5Literate(grpid, H5_INDEX_CRT_ORDER, H5_ITER_INC, &idx, op_func, 
                                                                      ^~~~~~~
/Users/wfisher/environments/local-serial-1.12.2-ac-clang/include/H5Lpublic.h:1835:42: note: passing argument to parameter 'op' here
                          H5L_iterate1_t op, void *op_data);
```

Works with: Apple clang version 15.0.0 (clang-1500.1.0.2.5)
Fails with: Apple clang version 15.0.0 (clang-1500.3.9.4)

--- a/h5_test/tst_h_files4.c
+++ b/h5_test/tst_h_files4.c
@@ -54,8 +54,9 @@ op_func (hid_t g_id, const char *name,
    strcpy((char *)op_data, name);
 #if H5_VERSION_LE(1, 10, 11) || defined(H5_USE_110_API_DEFAULT) || defined(H5_USE_18_API_DEFAULT) || defined(H5_USE_16_API_DEFAULT)
    /* This library is either 1.10.11 (the last 1.10 release) or earlier
-    * OR this a later version of the library built with a 1.10 or
-    * earlier API.
+    * OR this is a later version of the library built with a 1.10 or
+    * earlier API (earlier versions did not define their own USE
+    * API symbol).
     */
    if ((id = H5Oopen_by_addr(g_id, info->u.address)) < 0) ERR;
 #else

--- a/h5_test/tst_h_files4.c
+++ b/h5_test/tst_h_files4.c
@@ -45,21 +45,18 @@ with the H5Lvisit function call
 */
 herr_t
 op_func (hid_t g_id, const char *name, 
-#if H5_VERSION_GE(1,12,0)
-         const H5L_info2_t *info,
-#else
-         const H5L_info_t *info,
-#endif
+     const H5L_info_t *info,
 	 void *op_data)  
 {
    hid_t id;
    H5I_type_t obj_type;
 
    strcpy((char *)op_data, name);
-#if H5_VERSION_GE(1,12,0)
-   if ((id = H5Oopen_by_token(g_id, info->u.token)) < 0) ERR;
-#else
+#if defined(H5_USE_110_API_DEFAULT) || defined(H5_USE_18_API_DEFAULT) || defined(H5_USE_16_API_DEFAULT)
    if ((id = H5Oopen_by_addr(g_id, info->u.address)) < 0) ERR;
+#else
+   /* HDF5 1.12 switched from addresses to tokens to better support the VOL */
+   if ((id = H5Oopen_by_token(g_id, info->u.token)) < 0) ERR;
 #endif
 
 /* Using H5Ovisit is really slow. Use H5Iget_type for a fast

--- a/h5_test/tst_h_files4.c
+++ b/h5_test/tst_h_files4.c
@@ -52,7 +52,11 @@ op_func (hid_t g_id, const char *name,
    H5I_type_t obj_type;
 
    strcpy((char *)op_data, name);
-#if defined(H5_USE_110_API_DEFAULT) || defined(H5_USE_18_API_DEFAULT) || defined(H5_USE_16_API_DEFAULT)
+#if H5_VERSION_LE(1, 10, 11) || defined(H5_USE_110_API_DEFAULT) || defined(H5_USE_18_API_DEFAULT) || defined(H5_USE_16_API_DEFAULT)
+   /* This library is either 1.10.11 (the last 1.10 release) or earlier
+    * OR this a later version of the library built with a 1.10 or
+    * earlier API.
+    */
    if ((id = H5Oopen_by_addr(g_id, info->u.address)) < 0) ERR;
 #else
    /* HDF5 1.12 switched from addresses to tokens to better support the VOL */

